start-deployBukkitSpigot

```diff
@@ -1,24 +1,26 @@
 #!/bin/bash
 
+. /start-utils
+
 set -e
 
 function buildSpigotFromSource {
-  echo "Building Spigot $VANILLA_VERSION from source, might take a while, get some coffee"
+  log "Building Spigot $VANILLA_VERSION from source, might take a while, get some coffee"
   rm -rf /data/temp
   mkdir /data/temp
   cd /data/temp
 
   jvmOpts="-Xms${INIT_MEMORY:-$MEMORY} -Xmx${MAX_MEMORY:-$MEMORY}"
 
   curl -sSL -o /data/temp/BuildTools.jar https://hub.spigotmc.org/jenkins/job/BuildTools/lastSuccessfulBuild/artifact/target/BuildTools.jar && \
-    java $jvmOpts -jar /data/temp/BuildTools.jar --rev $VANILLA_VERSION 2>&1 |tee /data/spigot_build.log| while read l; do echo -n .; done; echo "done"
+    java $jvmOpts -jar /data/temp/BuildTools.jar --rev $VANILLA_VERSION 2>&1 |tee /data/spigot_build.log| while read l; do log -n .; done; log "done"
```

Comment:
Do we want this first log to be a log or echo? I'm assuming it's only for logging the build process, but I find it a little hard to read.

Reply:
As you suspected it’s just for showing a dot per line of the verbose build output. I think `-n` will carry over fine which is there to put all the dots on one long line. However that’ll means you should probably start with a `log -n ‘’` and then keep the echo’s as they are. That’ll provide the one logging prefix and append dots to that line.

Reply:
I tried a few solutions and landed at 633b482. It requires adding flag parsing to the `log` function. I'm not quite sure if I'm very fond of the solution. What's your thoughts?

Reply:
Oh yeah, I see what you mean. I'm ok with your current approach. The only other idea I could think of would be to introduce a `logn` variant that does the `echo -n`. I'm not totally liking that either, but at least that would result in two little functions rather than one. more complicated, function.

Reply:
I could any. I you are okay with the `logn` variant I think I'd prefer it. Makes it more clear what's going on, especially for someone who isn't very used to bash. And it's probably less error prone.

Reply:
Yes, please go with a `logn`. We’re almost there! :)